src/main/java/org/fcrepo/migration/validator/impl/ValidatingObjectHandler.java

```diff
@@ -220,6 +235,59 @@ public void validateDatastream(final String dsId, final ObjectReference objectRe
         }
     }
 
+    /**
+     * Validate the checksum of a datastream.
+     *
+     * This can fail in multiple ways:
+     * 1 - The F3 datastream can not be read
+     * 2 - The F6 headers do not contain a checksum (only checking sha512 atm)
+     * 3 - The two calculated checksums do not match
+     *
+     * @param dsVersion the datastream
+     * @param headers the ocfl headers
+     * @param version the version number
+     * @param builder helper for building ValidationResults
+     */
+    private void validateChecksum(final DatastreamVersion dsVersion, final ResourceHeaders headers,
+                                  final String version, final ValidationResultBuilder builder) {
+        final var success = "%s binary checksums match: %s";
+        final var error = "%s binary checksums do no match: sourceValue=%s, targetValue=%s";
+        final var notFound = "%s binary checksum not found in Fedora 6 headers";
+        final var exception = "%s binary checksum was unable to be calculated: exception=%s";
+
+        final HashCode sourceHash;
+        final var controlGroup = F3ControlGroup.fromString(dsVersion.getDatastreamInfo().getControlGroup());
+        if (checksum && controlGroup == F3ControlGroup.MANAGED) {
+            try {
+                // compute the checksum of the datastream
+                final var hasher = digestAlgorithm.hasher();
+                ByteStreams.copy(dsVersion.getContent(), Funnels.asOutputStream(hasher));
+                sourceHash = hasher.hash();
+            } catch (IOException e) {
+                validationResults.add(builder.fail(BINARY_METADATA, format(exception, version, e.toString())));
```

Comment:
@mikejritter :  Can you add a new ValidationType BINARY_CHECKSUM and use it here (as well as  below).  It will make it easier for folks to parse out what they're interested in in the TSV output.